lib/ansible/cli/galaxy.py

```diff
@@ -321,6 +490,56 @@ def execute_install(self):
         uses the args list of roles to be installed, unless -f was specified. The list of roles
         can be a name (which will be downloaded via the galaxy API and github), or it can be a local tar archive file.
         """
+        if context.CLIARGS['type'] == 'collection':
+            collections = context.CLIARGS['args']
+            force = context.CLIARGS['force']
+            output_path = context.CLIARGS['collections_path']
+            # TODO: use a list of server that have been configured in ~/.ansible_galaxy
+            servers = [context.CLIARGS['api_server']]
+            ignore_certs = context.CLIARGS['ignore_certs']
+            ignore_errors = context.CLIARGS['ignore_errors']
+            requirements_file = context.CLIARGS['requirements']
+            no_deps = context.CLIARGS['no_deps']
+            force_deps = context.CLIARGS['force_with_deps']
+
+            if collections and requirements_file:
+                raise AnsibleError("The positional collection_name arg and --requirements-file are mutually exclusive.")
+            elif not collections and not requirements_file:
+                raise AnsibleError("You must specify a collection name or a requirements file.")
+
+            if requirements_file:
+                requirements_file = GalaxyCLI._resolve_path(requirements_file)
+                collection_requirements = parse_collections_requirements_file(requirements_file)
+            else:
+                collection_requirements = []
+                for collection_input in collections:
+                    if ':' in collection_input:
+                        name, requirement = collection_input.split(':', 1)
+                    else:
+                        name = collection_input
+                        requirement = '*'
+                    collection_requirements.append((name, requirement, None))
+
+            output_path = GalaxyCLI._resolve_path(output_path)
+            collections_path = C.COLLECTIONS_PATHS
+
+            if len([p for p in collections_path if p.startswith(output_path)]) == 0:
```

Comment:
No need for an intermediate list in memory.
```suggestion
            if len(p for p in collections_path if p.startswith(output_path)) == 0:
```

Reply:
Seems like there is on Python 2.7, fails without it.